### PR TITLE
Fix persistent cache on windows

### PR DESCRIPTION
**Pull request by zhli1142015**
Summary:
Persistent cache feature caused rocks db crash on windows. I posted a issue for it, #6919. I found this is because no "persistent_cache_key_prefix" is generated for persistent cache. Looking repo history, "GetUniqueIdFromFile" is not implemented on Windows. So my fix is adding "NewId()" function in "persistent_cache" and using it to generate prefix for persistent cache. In this PR, i also re-enable related test cases defined in "db_test2" and "persistent_cache_test" for windows.
Test Plan:
1. run related test cases in "db_test2" and "persistent_cache_test" on windows and see it passed.
2. manually run db_bench.exe with "read_cache_path" and verified.

### CODE REVIEW

File: utilities/persistent_cache/persistent_cache_tier.h [hunk 1]
```diff
@@ -266,6 +266,8 @@ class PersistentCacheTier : public PersistentCache {
 
   virtual std::string GetPrintableOptions() const override = 0;
 
+  virtual uint64_t NewId() override;
```
Comment (cheng-chang) [inline utilities/persistent_cache/persistent_cache_tier.h:269]: does `NewId` override any interface?

Comment (zhli1142015) [reply utilities/persistent_cache/persistent_cache_tier.h:269]: yes, it's defined in "PersistentCache".

File: include/rocksdb/persistent_cache.h [hunk 2]
```diff
@@ -56,6 +56,8 @@ class PersistentCache {
   virtual StatsType Stats() = 0;
 
   virtual std::string GetPrintableOptions() const = 0;
+
+  virtual uint64_t NewId() = 0;
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply include/rocksdb/persistent_cache.h:64]: added, thanks for reviewing.

File: table/block_based/block_based_table_reader.h [hunk 3]
```diff
@@ -440,6 +440,9 @@ class BlockBasedTable : public TableReader {
                                   char* buffer, size_t* size);
   static void GenerateCachePrefix(Cache* cc, FSWritableFile* file, char* buffer,
                                   size_t* size);
+  static void GeneratePersistentCachePrefix(PersistentCache* cc,
```
Comment (riversand963) [inline table/block_based/block_based_table_reader.h:443]: The function name `GeneratePersistentCachePrefix` can be `GenerateCachePrefix`. We know it's for persistent cache by looking at its first argument. C++ function overloading will take care of it. The implementation of this function does not have anything specific to persistent cache.
One step forward is to use a template function for all the `GeneratePersistentCachePrefix` functions.

Comment (zhli1142015) [reply table/block_based/block_based_table_reader.h]: Good idea, updated, thanks.

**Comment by zhli1142015** [comment]
@riversand963 , @cheng-chang , could you please help to merge this pr?

**Pull request closed without merging**